src/router/storeSync/storeSync.routerPlugin.js

```diff
@@ -65,8 +65,8 @@ function storeMutationWatcher(store, mutation, router) {
                     routeChangeIsTriggeredByThisModule = false
                 })
             // if the short linked version of the URL is already defined,
-            // we must refresh it as the URL has just changed
-            if (store.state.position.shortLink) {
+            // and the menu opened, we must refresh it as the URL has just changed
+            if (store.getters.showMenu && store.state.share.keepUpdatingShortLink) {
```

Comment:
As all the information comes from the store you could move the check into the `generateShortLink` action.

Reply:
yes and no, `store.getters.showMenu` comes from `ui.store.js` and `store.state.share.keepUpdatingShortLink` from `share.store.js`. The philosophy we try to follow is that any cross store module interaction should be done in plugins 😉